be/src/service/backend_base.cpp

```diff
@@ -54,14 +52,28 @@
 
 namespace starrocks {
 
-using apache::thrift::TException;
-using apache::thrift::TProcessor;
-using apache::thrift::TMultiplexedProcessor;
-using apache::thrift::transport::TTransportException;
 using apache::thrift::concurrency::ThreadFactory;
 
 BackendServiceBase::BackendServiceBase(ExecEnv* exec_env) : _exec_env(exec_env) {}
 
+template <class Service>
+std::unique_ptr<ThriftServer> BackendServiceBase::create(ExecEnv* exec_env, int port) {
+    auto handler = std::make_shared<Service>(exec_env);
+    // TODO: do we want a BoostThreadFactory?
+    // TODO: we want separate thread factories here, so that fe requests can't starve
+    // cn requests
+    auto thread_factory = std::make_shared<ThreadFactory>();
+    auto processor = std::make_shared<BackendServiceProcessor>(handler);
+
+    LOG(INFO) << "StarRocksInternalService listening on " << port;
+    // TODO: May be rename be_service_threads to thrift_service_threads ?
+    return std::make_unique<ThriftServer>("BackendService", processor, port, exec_env->metrics(),
```

Comment:
Could add a trait method like `Service::name()` for getting the actual service name